harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,840 @@
+
+        # SSH to vlan ip address and execute command
+        _stdout, _stderr = self.ssh_client(
+            client, vlan_ip, vm_credential["user"], vm_credential["password"], 'ls', wait_timeout)
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find("bin") == 0, (
+            'Failed to ssh to VM management IP %s' % (vlan_ip))
+
+        # cleanup vm
+        delete_vm(api_client, unique_name, wait_timeout)
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="vlan_mgmt_connection",
+                            depends=["mgmt_network_connection"])
+    def test_vlan_to_mgmt_connection(self, api_client, request, client, unique_name,
+                                     vlan_network):
+        """
+        Manual test plan reference:
+        https://harvester.github.io/tests/manual/network/edit-network-form-change-management-to-vlan/
+
+
+        Steps:
+        1. Create an external VLAN network
+        2. Create a new VM
+        3. Make sure that the network is set to the vlan network with bridge as the type
+        4. Wait until the VM boot in running state
+        5. Edit VM and change from external VLAN to management network
+        6. Check VM should save and reboot
+        7. Check can ping the VM on the management network
+        """
+
+        wait_timeout = request.config.getoption('--wait-timeout')
+        client_ip = request.config.getoption('--endpoint').strip('https://')
+
+        tcp = "sudo sed -i 's/AllowTcpForwarding no/AllowTcpForwarding yes/g' /etc/ssh/sshd_config"
+        restart_ssh = "sudo systemctl restart sshd.service"
+
+        self.ssh_client(client, client_ip, node_user, node_password,
+                        tcp, wait_timeout)
+
+        self.ssh_client(client, client_ip, node_user, node_password,
+                        restart_ssh, wait_timeout)
+
+        spec = api_client.vms.Spec(1, 2, mgmt_network=False)
+        spec.user_data += cloud_user_data.format(password=vm_credential["password"])
+        unique_name = unique_name + "-vlan-mgmt"
+
+        # Create VM
+        spec.add_image(image_name, "default/" + image_name)
+        spec.add_network("default", vlan_network['id'])
+
+        code, data = api_client.vms.create(unique_name, spec)
+        assert 201 == code, (f"Failed to create vm with error: {code}, {data}")
+
+        # Check VM start in running state
+        check_vm_running(api_client, unique_name, wait_timeout)
+
+        # Check until VM ip address exists
+        check_vm_ip_exists(api_client, unique_name, wait_timeout)
+
+        # get data from running VM and transfer to spec
+        code, data = api_client.vms.get(unique_name)
+        spec = spec.from_dict(data)
+
+        # Switch to vlan network
+        spec.mgmt_network = True
+
+        # Update VM spec
+        code, data = api_client.vms.update(unique_name, spec)
+
+        spec.delete_vlan_network(spec, vlan_network['id'])
```

Comment:
steps in here is not align to test plan, we would expect the VM will have only 1 network.
so we should remove mgmt and add VLAN in the same time, it would be:
```python
        # Switch to vlan network
        spec.mgmt_network = True

        # Update VM spec   -------- don't update VM otherwise the VM will have 2 NICs
        # code, data = api_client.vms.update(unique_name, spec)

        spec.delete_vlan_network(spec, vlan_network['id'])
```

and it could be simplify as:
```python
spec.networks = []  # clear all NICs
spec.mgmt_network = True
```